modules/apps/29-fee/keeper/keeper.go

```diff
@@ -189,6 +174,59 @@ func (k Keeper) GetAllRelayerAddresses(ctx sdk.Context) []*types.RegisteredRelay
 	return registeredAddrArr
 }
 
+// SetForwardRelayerAddress sets the forward relayer address during OnRecvPacket in case of async acknowledgement
+func (k Keeper) SetForwardRelayerAddress(ctx sdk.Context, packetId *channeltypes.PacketId, address string) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyForwardRelayerAddress(packetId), []byte(address))
+}
+
+// GetForwardRelayerAddress gets forward relayer address for a particular packet
+func (k Keeper) GetForwardRelayerAddress(ctx sdk.Context, packetId *channeltypes.PacketId) (string, bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeyForwardRelayerAddress(packetId)
+	if !store.Has(key) {
+		return "", false
+	}
+
+	addr := string(store.Get(key))
+	return addr, true
+}
+
+// GetAllForwardRelayerAddresses returns all forward relayer addresses stored for async acknowledgements
+func (k Keeper) GetAllForwardRelayerAddresses(ctx sdk.Context) []*types.ForwardRelayerAddress {
```

Comment:
could you add a unit test for this? A followup pr is fine

Reply:
I'll create an issue. Thanks